packages/testing/src/test/chai-retry-plugin.unit.ts

```diff
@@ -0,0 +1,295 @@
+import Chai, { expect } from 'chai';
+import { chaiRetryPlugin } from '../chai-retry-plugin/chai-retry-plugin';
+
+Chai.use(chaiRetryPlugin);
+
+describe('chai-retry-plugin', () => {
+    it('should retry a function that eventually succeeds', async () => {
+        let attempts = 0;
+
+        const funcToRetry = () => {
+            attempts++;
+            if (attempts < 3) {
+                throw new Error('Failed');
+            }
+            return 'Success';
+        };
+
+        await expect(funcToRetry).to.retry({ retries: 5, delay: 10 });
+        expect(attempts).to.equal(3);
+    });
+
+    it('should retry a function that always fails and throw an error', async () => {
+        let attempts = 0;
+
+        const funcToRetry = () => {
+            attempts++;
+            throw new Error('Failed');
+        };
+
+        try {
+            await expect(funcToRetry).to.retry({ retries: 3 });
+
+            throw new Error('This should not be called');
+        } catch (error) {
+            expect(error).to.be.an.instanceOf(Error);
+            expect((error as Error).message).to.includes('Limit of 3 retries');
+            expect(attempts).to.equal(3);
+        }
+    });
+
+    describe('options should work correctly:', () => {
+        it('timeout after the specified duration', async () => {
+            const funcToRetry = async () => {
+                await new Promise((resolve) => setTimeout(resolve, 500));
+                return 'Success';
+            };
+
+            await expect(funcToRetry)
+                .retry({ retries: 2, timeout: 250 })
+                .then(
```

Comment:
why not `try ... catch`?

Reply:
@yurii-ve just made it like that 😅 but yeah, could be reworked